src/lib/analytics.ts

```diff
@@ -0,0 +1,79 @@
+import {
+    init as amplitudeInit,
+    track as amplitudeTrack,
+} from '@amplitude/analytics-node';
+import machine from 'node-machine-id';
+import { getAddress } from './arweaveHelper';
+import { withAsync } from './withAsync';
+
+const AMPLITUDE_TRACKING_ID = '92a463755ed8c8b96f0f2353a37b7b2';
+const PLATFORM = '@protocol.land/git-remote-helper';
+
+let isInitialized = false;
+
+const initializeAmplitudeAnalytics = async () => {
+    if (isInitialized) return;
+
+    await amplitudeInit(AMPLITUDE_TRACKING_ID).promise;
+    isInitialized = true;
+};
+
+export const trackAmplitudeAnalyticsEvent = async (
+    category: string,
+    action: string,
+    label: string,
+    data?: Record<any, any>
+) => {
+    try {
+        let eventOptions: { user_id?: string; device_id?: string } = {
+            user_id: undefined,
+            device_id: undefined,
+        };
+
+        await initializeAmplitudeAnalytics();
+
+        const { response: userAddress } = await withAsync(() => getAddress());
+        if (userAddress) {
+            eventOptions = { user_id: userAddress };
+        } else {
+            const { response: machineId } = await withAsync(() =>
+                machine.machineId(true)
+            );
+            if (machineId) {
+                eventOptions = { device_id: machineId };
+            }
+        }
+        if (eventOptions?.user_id || eventOptions?.device_id) {
+            await amplitudeTrack(
+                category,
+                {
+                    action,
+                    label,
+                    platform: PLATFORM,
+                    ...data,
+                },
+                eventOptions
+            ).promise;
```

Comment:
i dont think its worth it to block the thread by awaiting track events. Thoughts @7i7o ?

Reply:
I agree. Specially when we ignore any errors that events tracking might throw.

Reply:
@kranthicodes I tested this.
Even if we don't await event tracking, the process waits until the node execution ends to return control of the console, so the process will await for the promise to resolve anyway.